### DISCUSSION TITLE
Debug TypeScript ESLint paths resolution

### DISCUSSION THREAD

**Discussion by osdiab** [zed-industries/zed#6977] [Feature Requests]
Hi, I normally use VSCode and considering a jump to Zed, but out of box it is highlighting an import to a path configured via the TypeScript `paths` property as a problem because it is not resolvable according to ESLint (but not according to TypeScript).

<img width="735" alt="Screenshot 2024-01-29 at 18 47 38" src="https://github.com/zed-industries/zed/assets/1890926/4d394109-9087-4acb-98bc-c73cdd3906c5">

How would one go about debugging this Zed behavior? Perhaps the LSP for ESLint isn't actually catching my `.eslintrc.js`? Thanks!

**Comment by SomeoneToIgnore** [comment]
Hi, the best way so far would be
* LSP logs (debug: open language server logs in the action panel)
* Zed logs for the large calamities like LSP server did not start at all

Another way would be to add more debug logs (and, maybe, even turn those into a PR later) and run a debug version of Zed to see what's happening.
https://github.com/zed-industries/zed/pull/3457 was the last attempt to adjust eslint diagnostics and could serve as a reference where to log things.

**Comment by osdiab** [comment]
Looks like the LSP logs don't have much info:

```
ESLint library loaded from:
***@***.***
/node_modules/eslint/lib/api.js
stderr: Warning: React version was set to "detect" in eslint-plugin-react
settings, but the "react" package is not installed. Assuming latest React
version for glinting.
```

Not sure where it's getting the ESLint config from or anything. So yeah I
guess the LSP logs are just not very thorough yet. I'll probably come back
to Zed once it's had a bit more time to mature, thanks!

Omar

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
